ingestion/src/metadata/ingestion/source/vertica.py

```diff
@@ -9,9 +9,170 @@
+    if schema is not None:
+        schema_condition = "lower(table_schema) = '%(schema)s'" % {
+            "schema": schema.lower()
+        }
+    else:
+        schema_condition = "1"
+
+    s = sql.text(
+        dedent(
+            VERTICA_GET_COLUMNS.format(
+                table=table_name.lower(), schema_condition=schema_condition
+            )
+        )
+    )
+
+    spk = sql.text(
+        dedent(
+            VERTICA_GET_PRIMARY_KEYS.format(
+                table=table_name.lower(), schema_condition=schema_condition
+            )
+        )
+    )
+
+    pk_columns = [x[0] for x in connection.execute(spk)]
+    columns = []
+    for row in connection.execute(s):
+        name = row.column_name
+        dtype = row.data_type.lower()
+        primary_key = name in pk_columns
+        default = row.column_default
+        nullable = row.is_nullable
+        comment = row.comment
+
+        column_info = self._get_column_info(
+            name,
+            dtype,
+            default,
+            nullable,
+            schema,
+            comment,
+        )
+        column_info.update({"primary_key": primary_key})
+        columns.append(column_info)
+    return columns
+
+
+def _get_column_info(
+    self,
+    name,
+    format_type,
+    default,
+    nullable,
+    schema,
+    comment,
+):
+
+    # strip (*) from character varying(5), timestamp(5)
+    # with time zone, geometry(POLYGON), etc.
+    attype = re.sub(r"\(.*\)", "", format_type)
+
+    charlen = re.search(r"\(([\d,]+)\)", format_type)
+    if charlen:
+        charlen = charlen.group(1)
+    args = re.search(r"\((.*)\)", format_type)
+    if args and args.group(1):
+        args = tuple(re.split(r"\s*,\s*", args.group(1)))
+    else:
+        args = ()
+    kwargs = {}
+
+    if attype == "numeric":
+        if charlen:
+            prec, scale = charlen.split(",")
+            args = (int(prec), int(scale))
+        else:
+            args = ()
+    elif attype == "integer":
+        args = ()
+    elif attype in ("timestamptz", "timetz"):
+        kwargs["timezone"] = True
+        if charlen:
+            kwargs["precision"] = int(charlen)
+        args = ()
+    elif attype in (
+        "timestamp",
+        "time",
+    ):
+        kwargs["timezone"] = False
+        if charlen:
+            kwargs["precision"] = int(charlen)
+        args = ()
+    elif attype.startswith("interval"):
+        field_match = re.match(r"interval (.+)", attype, re.I)
+        if charlen:
+            kwargs["precision"] = int(charlen)
+        if field_match:
+            kwargs["fields"] = field_match.group(1)
+        attype = "interval"
+        args = ()
+    elif charlen:
+        args = (int(charlen),)
+
+    while True:
+        if attype.upper() in self.ischema_names:
+            coltype = self.ischema_names[attype.upper()]
+            break
+        else:
+            coltype = None
+            break
```

Comment:
Hey @ulixius9, according to sonarcloud, this loop shows up as a bug. Looking at the same, I think we won't need the loop as it will only loop once at initial stage and break out of the loop. we can remove the loop, can you test the same after removing it?